splunk_event/test_splunk_event.py

```diff
@@ -184,22 +184,24 @@ def test_not_dispatch_sids_checks(self, mocked_splunk_helper):
         second_persistent_data = self.check.status.data.get(instance.get('url') + "minimal_events")
         # The second run_check will finalize the previous saved search ids and create a new one,
         # so we make sure this is the case
-        self.assertNotEqual(first_persistent_data[0][1], second_persistent_data[0][1])
+        self.assertEqual(first_persistent_data, second_persistent_data)
 
-        mocked_splunk_helper.return_value.finalize_sid = mock.MagicMock(return_value=404)
+        mocked_splunk_helper.return_value.finalize_sid = mock.MagicMock(side_effect=FinalizeException(500,"Internal error"))
 
-        # Run the check 3rd time and get the persistent status data
-        self.run_check(config, mocks={
-            '_search': _mocked_search,
-            '_saved_searches': _mocked_saved_searches,
-            '_auth_session': _mocked_auth_session
-        }, force_reload=True)
+        thrown = False
+        try:
+            self.run_check(config, mocks={
+                '_search': _mocked_search,
+                '_saved_searches': _mocked_saved_searches,
+                '_auth_session': _mocked_auth_session
+            }, force_reload=True)
+        except CheckException:
+            thrown = True
 
-        third_persistent_data = self.check.status.data.get(instance.get('url') + "minimal_events")
+        self.assertTrue(thrown)
```

Comment:
what the persistent state look like now?

Reply:
something like this - `{'http://localhost:8089/minimal_metrics': 'minimal_metrics'}`

Reply:
lets expect that, if finalize fails persistent state should still be there

Reply:
agree. it will be there.